components/reusable-components/pagination/HomePagination.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from 'react';
+import { fetchItems, Item } from './lib/data';
+import StructurePagination from './StructurePagination';
+import Dropdown from './Dropdown';
+import styles from './HomePagination.module.scss';
+
+const HomePagination: React.FC = () => {
+  const [items, setItems] = useState<Item[]>([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalPages, setTotalPages] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState(25);
+
+  const options = [25, 50, 100];
+
+  useEffect(() => {
+    const loadItems = async () => {
+      const data = await fetchItems(currentPage, itemsPerPage);
+      setItems(data.items);
+      setTotalPages(Math.ceil(data.total / itemsPerPage));
+    };
+
+    loadItems();
+  }, [currentPage, itemsPerPage]);
+
+  useEffect(() => {
+    setCurrentPage(1);
+  }, [itemsPerPage]);
```

Comment:
consider the following scenario here:
we have 100 users, we are showing 25 per page, user is on page 4, and changes in the dropdown to 50 per page. So now we only have 2 pages, and not 4.
How can we handle that scenario?

Reply:
@aleksandarmicev also please consider the scenario in the comment above. Please tag me when this is done